bigquery/system-test/datasets.test.js

```diff
@@ -0,0 +1,85 @@
+// Copyright 2016, Google, Inc.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+var BigQuery = require('@google-cloud/bigquery');
+var uuid = require('node-uuid');
+var program = require('../datasets');
+
+var bigquery = BigQuery();
+var projectId = process.env.GCLOUD_PROJECT;
+var datasetId = 'nodejs-docs-samples-test-' + uuid.v4();
+
+// BigQuery only accepts underscores
+datasetId = datasetId.replace(/-/gi, '_');
+
+describe('bigquery:datasets', function () {
+  after(function (done) {
+    bigquery.dataset(datasetId).delete({
+      force: true
+    }, function () {
+      // Ignore any error, the dataset might already have been successfully deleted
```

Comment:
Can't, otherwise I wouldn't be able to prevent an error from being passed to `done` (which was the point).